### DISCUSSION TITLE
Twitter notification with StockAlert

### DISCUSSION THREAD

**Discussion by xmpierce1** [Prince25/StockAlertBot#69] [Requests]
it would be great to have native support to have a Twitter notification via Twitter API direct from StockAlert similar to Streetmerchant.

**Comment by Prince25** [comment]
I've received this request before on Discord. Twitter, in essence, is meant to notify other users (the followers), so it doesn't really align with the goal of this program. It just doesn't make sense to send a tweet alerting others, when you yourself, will not be alerted. Plus, I believe it would enable the wrong type of crowd to gain followers and take advantage of their situation. It's for the same reason I won't add auto-checkout support. I want this program to be powered by the user, and solely, for the user. Nevertheless, thank you for your suggestion. I'm always open to hear more.

Since I have never used Street Merchant, how easy is Street Merchant to use? All those options and configurations seems like it would overwhelm the average user, especially when they would have to do it all via a text editor.

**Comment by xmpierce1** [comment]
Discord notify users as well.  I understand your point, but the followers are trying to get items themselves without reselling, but you will have resellers for sure. Street Merchant has several methods of notifications, with Twitter as just one of them. It does not seem difficult, but you have to mess with the environment file a lot. I like your code because of the simple UI interface, especially the ability to add products.  

 

Just my opinion, I do like StockAlert and it works very well.

**Comment by Prince25** [comment]
I like take your suggestion into consideration, especially if more people request it. Right now, my priority is recoding everything for a much more unstable and easier development experience. If you have any developers you know, send them my way. Haha. Wish I had 200+ contributors as Street Merchant does. lol

**Reply by xmpierce1** [reply]
I did get a solution for Twitter notification that is not expensive. Use StockAlert to post to Discord and then use Integromat to create a connector from Discord to Twitter. its a cheap solution and works versus Zapier.

**Reply by Prince25** [reply]
Thanks for looking into it. What is the latency on that? I'd expect it to be delayed.

**Comment by xmpierce1** [comment]
One minute max..

Sent from my mobile device


On Apr 29, 2021, at 12:17 AM, Prabhjot Singh ***@***.***> wrote:

﻿

Thanks for looking into it. What is the latency on that? I'd expect it to be delayed.

**Reply by Prince25** [reply]
Wow. Perhaps a better solution is to just implement the Twitter API and put the secrets in the `.env`.

**Reply by xmpierce1** [reply]
I like to see if I can use the IFTTL webhook to Twitter and see what happens. The API is a good option for sure using these variable settings. 

TWITTER_ACCESS_TOKEN_KEY=
TWITTER_ACCESS_TOKEN_SECRET=
TWITTER_CONSUMER_KEY=
TWITTER_CONSUMER_SECRET=
TWITTER_TWEET_TAGS=
